controllers/hcloudmachine_controller.go

```diff
@@ -294,3 +301,57 @@ func (r *HCloudMachineReconciler) HetznerClusterToHCloudMachines(_ context.Conte
 		return result
 	}
 }
+
+func IgnoreHetznerClusterConditionUpdates(logger logr.Logger) predicate.Funcs {
+	return predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			log := logger.WithValues(
+				"predicate", "IgnoreHetznerClusterConditionUpdates",
+				"type", "update",
+				"namespace", e.ObjectNew.GetNamespace(),
+				"kind", strings.ToLower(e.ObjectNew.GetObjectKind().GroupVersionKind().Kind),
+				"name", e.ObjectNew.GetName(),
+			)
+
+			var oldCluster, newCluster *infrav1.HetznerCluster
+			var ok bool
+			// This predicate only looks at HetznerCluster objects
+			if oldCluster, ok = e.ObjectOld.(*infrav1.HetznerCluster); !ok {
+				return true
+			}
+			if newCluster, ok = e.ObjectNew.(*infrav1.HetznerCluster); !ok {
+				// Something weird happened, and we received two different kinds of objects
+				return true
+			}
+
+			// We should not modify the original objects, this causes issues with code that relies on the original object.
+			oldCluster = oldCluster.DeepCopy()
+			newCluster = newCluster.DeepCopy()
+
+			// Set fields we do not care about to nil
+
+			oldCluster.ManagedFields = nil
+			newCluster.ManagedFields = nil
+
+			oldCluster.ResourceVersion = ""
+			newCluster.ResourceVersion = ""
+
+			oldCluster.Status.Conditions = nil
+			newCluster.Status.Conditions = nil
+
+			if reflect.DeepEqual(oldCluster, newCluster) {
+				// Only insignificant fields changed, no need to reconcile
+				log.V(4).Info("Update to resource only changes insignificant fields, will not enqueue event")
```

Comment:
we don't use V(4) and higher anywhere. If you wanna include it, then use V(1).

Reply:
:+1: Copied those levels from the "not-paused" predicate in the CAPI repo. Will update in the next push.

Reply:
I do not think these need to be in the default logging level, but its helpful if one can enable these through a flag. Whats the level I should use for that?

Reply:
IIUC
- `V(1)` equals the Zap debug logging, which is the default in caph
- `V(0)` would be Zap info logging

I set both the `V(1)` and rebased.